plugins/hydra_nevergrad_sweeper/tests/test_nevergrad_sweeper_plugin.py

```diff
@@ -6,45 +6,92 @@
 
 import nevergrad as ng
 import pytest
+from hydra.core.override_parser.overrides_parser import OverridesParser
 from hydra.core.plugins import Plugins
 from hydra.plugins.sweeper import Sweeper
 from hydra.test_utils.test_utils import TSweepRunner, chdir_plugin_root
 from omegaconf import DictConfig, OmegaConf
 
 from hydra_plugins.hydra_nevergrad_sweeper import core
+from hydra_plugins.hydra_nevergrad_sweeper.nevergrad_sweeper import NevergradSweeper
 
 chdir_plugin_root()
 
 
 def test_discovery() -> None:
-    assert core.NevergradSweeper.__name__ in [
+    assert NevergradSweeper.__name__ in [
         x.__name__ for x in Plugins.instance().discover(Sweeper)
     ]
 
 
 @pytest.mark.parametrize(  # type: ignore
-    "description,param_cls,value_cls",
+    "params,param_cls, param_val",
     [
-        ("blu,blublu", ng.p.Choice, str),
-        (["blu", "blublu"], ng.p.Choice, str),
-        ("0,1,2", ng.p.TransitionChoice, float),
-        ([0, 1, 2], ng.p.TransitionChoice, float),
-        ("int:0,1,2", ng.p.TransitionChoice, int),
-        ("str:0,1,2", ng.p.Choice, str),
-        ("0.0,12.0,2.0", ng.p.Choice, float),
-        ("int:1:12", ng.p.Scalar, int),
-        ("1:12", ng.p.Scalar, float),
-        ("log:0.01:1.0", ng.p.Log, float),
-        ("blublu", str, str),
+        ([1, 2, 3], ng.p.Choice, {1, 2, 3}),
+        (["1", "2", "3"], ng.p.Choice, {"1", "2", "3"}),
+        ({"lower": 1, "upper": 12, "log": True}, ng.p.Log, ([1.0], [12.0])),
+        ({"lower": 1, "upper": 12}, ng.p.Scalar, ([1.0], [12.0])),
+        ({"lower": 1, "upper": 12, "integer": True}, ng.p.Scalar, ([1], [12])),
+        (
+            {"lower": 1, "upper": 12, "log": True, "integer": True},
+            ng.p.Log,
+            ([1], [12]),
+        ),
     ],
 )
-def test_make_nevergrad_parameter(
-    description: Any, param_cls: Any, value_cls: Any
+def test_get_nevergrad_parameter_from_config(
+    params: str, param_cls: Any, param_val: Any
 ) -> None:
-    param = core.make_nevergrad_parameter(description)
-    assert isinstance(param, param_cls)
-    if param_cls is not str:
-        assert isinstance(param.value, value_cls)
+    param = core.get_nevergrad_parameter(params)
+    verify_param(param, param_cls, param_val)
+
+
+@pytest.mark.parametrize(  # type: ignore
+    "override,param_cls, param_val",
+    [
+        ("key=choice(1,2)", ng.p.Choice, {1, 2}),
```

Comment:
As i was working on the tests I realized why the tests was written in that way originally, the tl;dr is I think this is more straightforward 

The object looks something like the following, with a few default values that we are not setting.
```
TransitionChoice(choices=Tuple(hello,world,nevergrad),positions=Array{Cd(0,3)},transitions=[1. 1.]):world
```

Reply:
I understand that this was the original implementation. There should be a good reason to delegate the initialization to the test function like this and I am just not seeing it.
Please move to explicit initialization, it will make things much easier to understand.

Reply:
Agreed that explicitly init would make it easier to understand.

However as I pasted above there's a lot of extra fields in the object that we are not touching in the sweeper, we only have expectation for the values in the ng params. Also to explicitly init the object we'd need add dependency on `numpy`.

Reply:
Just initialize the exact same fields you are initializing now and let NG initialize the rest as defaults?
What you are doing is calling the `__init__` of say ng.p.Choice indirectly, passing multiple parameters.
It seems like it should be possible to do do the same explicitly, passing the same parameters.

Can you give an example of where this is not the case?

Reply:
You are right. I overthought it :) i've updated the tests accordingly.